### PR TITLE
Omit xml:space if not needed, while writing back SVG files.

### PR DESCRIPTION
**Pull request by H1Gdev**
ref. [this comment](https://github.com/vvvv/SVG/pull/471#pullrequestreview-242359019).

## sample SVG.

```svg
<?xml version="1.0" encoding="utf-8"?>
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" width="100" height="80">
<text fill="blue"  font-size="12" font-family="Verdana"                      x="10" y="20"> SVG  is  XML. </text>
<text fill="red"   font-size="12" font-family="Verdana" xml:space="default"  x="10" y="40"> SVG  is  XML. </text>
<text fill="green" font-size="12" font-family="Verdana" xml:space="preserve" x="10" y="60"> SVG  is  XML. </text>
</svg>
```

## output.svg

```svg
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg x="0" y="0" width="100" height="80" preserveAspectRatio="xMidYMid" font-size="0" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xml="http://www.w3.org/XML/1998/namespace" version="1.1">
  <text x="10" y="20" font-family="Verdana" font-size="12" style="fill:blue;"> SVG  is  XML. </text>
  <text xml:space="default" x="10" y="40" font-family="Verdana" font-size="12" style="fill:red;"> SVG  is  XML. </text>
  <text xml:space="preserve" x="10" y="60" font-family="Verdana" font-size="12" style="fill:green;"> SVG  is  XML. </text>
</svg>
```

### CODE REVIEW

File: Source/SvgElement.cs [hunk 1]
```diff
@@ -407,8 +417,8 @@ public string ID
         [SvgAttribute("space", SvgAttributeAttribute.XmlNamespace)]
         public virtual XmlSpaceHandling SpaceHandling
         {
-            get { return (this.Attributes["space"] == null) ? XmlSpaceHandling.@default : (XmlSpaceHandling)this.Attributes["space"]; }
-            set { this.Attributes["space"] = value; }
+            get { return GetAttribute("space", Inherited, XmlSpaceHandling.inherit); }
```
Comment (H1Gdev) [inline Source/SvgElement.cs]: Other properties also I think that there is more better to call this method.

File: Source/Document Structure/SvgFragment.cs [hunk 2]
```diff
@@ -148,6 +148,12 @@ public override string FontFamily
             set { this.Attributes["font-family"] = value; }
         }
 
+        public override XmlSpaceHandling SpaceHandling
+        {
+            get { return GetAttribute("space", false, XmlSpaceHandling.@default); }
```
Comment (H1Gdev) [inline Source/Document Structure/SvgFragment.cs]: Other properties also I think that there is more better to call this method.

File: Source/SvgElement.cs [hunk 3]
```diff
@@ -282,6 +282,16 @@ protected internal virtual SvgAttributeCollection Attributes
             }
         }
 
+        protected bool Inherited { get; set; } = true;
+
+        protected internal TAttributeType GetAttribute<TAttributeType>(string attributeName, bool inherit, TAttributeType defaultValue = default(TAttributeType))
+        {
+            if (inherit)
```
Comment (mrbean-bremen) [inline Source/SvgElement.cs]: Why not just use `Inherited` here instead of adding that extra parameter?

File: Source/SvgElement.cs [hunk 4]
```diff
@@ -609,65 +619,81 @@ protected virtual void WriteAttributes(XmlTextWriter writer)
...
-                                forceWrite = true;
+                                object parentValue;
+                                if (TryResolveParentAttributeValue(attr.Attribute.Name, out parentValue))
+                                {
+                                    if ((parentValue == propertyValue)
+                                        || ((parentValue != null) && parentValue.Equals(propertyValue)))
+                                        continue;
+
+                                    forceWrite = true;
+                                }
```
Comment (mrbean-bremen) [inline Source/SvgElement.cs]: The compare tool really does a bad job here...

Comment (H1Gdev) [reply Source/SvgElement.cs]: I think so, too...

File: Source/SvgElement.cs (hunk 3 above)
Comment (H1Gdev) [reply Source/SvgElement.cs]: Thank you for your review.

Some attributes are always `false` like SvgFragment, so add that extra parameter.

Comment (H1Gdev) [reply Source/SvgElement.cs]: Oh...
`svg` element can nest another `svg` element.

Comment (mrbean-bremen) [reply Source/SvgElement.cs]: Right. So with that in mind, I still think that this parameter is not needed - it is better in my opinion to make sure that the `Inherited` property is set correctly, and use that inside `GetAttribute`

Comment (H1Gdev) [reply Source/SvgElement.cs]: Please compare "id" attribute with "xml: space" attribute.

File: Source/SvgElement.cs [hunk 5]
```diff
@@ -393,7 +403,7 @@ protected RectangleF TransformedBounds(RectangleF bounds)
         [SvgAttribute("id")]
         public string ID
         {
-            get { return this.Attributes.GetAttribute<string>("id"); }
+            get { return GetAttribute<string>("id", false); }
```
Comment (H1Gdev) [inline Source/SvgElement.cs]: `inherit` argument is always `false` because "Id" attribute is not inherited.

File: Source/SvgElement.cs (hunk 3 above)
Comment (mrbean-bremen) [reply Source/SvgElement.cs]: Ok, thanks - I had some mind block, obviously. Sorry for the peckering...

**Pull request merged**